### PR TITLE
Get json ceremonies compat

### PR DESCRIPTION
**Pull request by ManjotS**

### CODE REVIEW

File: api/lib/getJSON.js [hunk 1]
```diff
@@ -45,6 +68,24 @@ const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
     verse.lineNo = row.LineNo;
     verse.updated = row.Updated;
     verse.visraam = JSON.parse(row.Visraam);
+    /* eslint-disable no-param-reassign */
+    if (Array.isArray(verse.visraam)) {
```
Comment (AkalUstat) [inline api/lib/getJSON.js:72]: would prefer this to be in a seperate function that you call, makes it more readable imo (ie if the function name was `compactVishraams` or something, its more declarative so more easier to understand when reading, ofc imo)

File: api/lib/getJSON.js [hunk 2]
```diff
@@ -45,6 +68,24 @@ const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
...
+        Object.keys(line).forEach(vtype => {
+          if (!accumulator[vtype]) accumulator[vtype] = [];
+          line[vtype].forEach(v => {
+            if (i > 0) v.p = parseInt(v.p, 10) + totalWordCount;
+            accumulator[vtype].push(v);
+          });
+        });
+        totalWordCount += wordCount[i];
+      });
+      verse.visraam = accumulator;
```
Comment (AkalUstat) [inline api/lib/getJSON.js:86]: Don't really like this mutation of parameters, would rather just have a return value

File: api/lib/getJSON.js [hunk 3]
```diff
@@ -1,5 +1,27 @@
+const object = require('lodash/fp/object');
+
+const reduceObj = (accumulator, currentObj) => {
```
Comment (AkalUstat) [inline api/lib/getJSON.js:3]: eventually would be nice to, instead of repeating this code across 3 codebases, maybe pull into a seperate lib?

Comment (ManjotS) [reply api/lib/getJSON.js:3]: if it keeps growing we will.. right now the lib dirs are too different

File: api/lib/getJSON.js [hunk 4]
```diff
@@ -1,5 +1,45 @@
...
+    }
+  });
+  return newAccumulator;
+};
+
+const concatObjects = array => {
+  return array.reduce((accumulator, current) => reduceObj(accumulator, current), {});
+};
+
+/* eslint-disable no-param-reassign */
```
Comment (AkalUstat) [inline api/lib/getJSON.js:22]: Probably don’t need this disable anymore but everything else lgtm👍

File: sql/sttm-api issue15 - wordcount_arrays.sql [hunk 5]
```diff
@@ -0,0 +1,117 @@
+create or replace view v_Ceremonies_Shabad as
```
Comment (tsingh777) [inline sql/sttm-api issue15 - wordcount_arrays.sql:1]: what's with this file name. are these like liquibase scripts to make db changes as we evolve?

File: api/lib/getJSON.js [hunk 6]
```diff
@@ -45,6 +86,8 @@ const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
     verse.lineNo = row.LineNo;
     verse.updated = row.Updated;
     verse.visraam = JSON.parse(row.Visraam);
+    const wordCount = JSON.parse(row.WordCount) || [0];
+    if (Array.isArray(verse.visraam)) verse.visraam = reduceVisraams(verse.visraam, wordCount);
```
Comment (tsingh777) [inline api/lib/getJSON.js:90]: nit: prefer `if` to have `{` `}`

File: api/lib/getJSON.js (hunk 4 above)
Comment (tsingh777) [inline api/lib/getJSON.js:22]: remove

File: api/lib/getJSON.js [hunk 7]
```diff
@@ -1,5 +1,45 @@
...
+  return array.reduce((accumulator, current) => reduceObj(accumulator, current), {});
+};
+
+/* eslint-disable no-param-reassign */
+const reduceVisraams = (visraam, wordCount) => {
+  let totalWordCount = 0;
+  const accumulator = {};
+  visraam.forEach((line, i) => {
+    Object.keys(line).forEach(vtype => {
+      if (!accumulator[vtype]) accumulator[vtype] = [];
```
Comment (tsingh777) [inline api/lib/getJSON.js:28]: nit: prefer if to have { }

File: api/lib/getJSON.js [hunk 8]
```diff
@@ -1,5 +1,45 @@
...
+      line[vtype].forEach(v => {
+        if (i > 0) v.p = parseInt(v.p, 10) + totalWordCount;
```
Comment (tsingh777) [inline api/lib/getJSON.js:30]: nit: prefer if to have { }

File: api/lib/getJSON.js [hunk 9]
```diff
@@ -1,5 +1,45 @@
+const object = require('lodash/fp/object');
+
+const reduceObj = (accumulator, currentObj) => {
```
Comment (tsingh777) [inline api/lib/getJSON.js:3]: some comments would be useful on why we need to to this. (we are going to forget some time down the line).

File: api/lib/getJSON.js [hunk 10]
```diff
@@ -1,5 +1,45 @@
...
+    totalWordCount += wordCount[i];
+  });
+  return accumulator;
+};
+/* eslint-enable no-param-reassign */
+
 const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
-  const translations = JSON.parse(row.Translations);
+  let translations = JSON.parse(row.Translations);
+  if (Array.isArray(translations)) translations = concatObjects(translations);
```
Comment (tsingh777) [inline api/lib/getJSON.js:42]: nit: prefer if to have { }

File: api/lib/getJSON.js [hunk 11]
```diff
@@ -32,7 +72,8 @@ const prepVerse = (row, includeMeta = false, liveSearch = 0) => {
   };
 
   if (liveSearch !== 1) {
-    const transliterations = JSON.parse(row.Transliterations);
+    let transliterations = JSON.parse(row.Transliterations);
+    if (Array.isArray(transliterations)) transliterations = concatObjects(transliterations);
```
Comment (tsingh777) [inline api/lib/getJSON.js:76]: nit: prefer if to have { }

File: sql/sttm-api issue15 - wordcount_arrays.sql (hunk 5 above)
Comment (ManjotS) [reply sql/sttm-api issue15 - wordcount_arrays.sql:1]: basically not having documented the history of the ddl well has caused issues, so now when I make ddl changes I want them committed to repo so its documented/traceable

File: api/lib/getJSON.js (hunk 4 above)
Comment (ManjotS) [reply api/lib/getJSON.js:22]: its required. it wont pass otherwise and the logic would have to be completely rewritten by writing some other loops/arrays

**Pull request merged**